src/problem/solve_problem.jl

```diff
@@ -2,25 +2,30 @@
 """
     initial_solution(
         problem::Problem;
-        disturbance_clusters::Set{Int64}=Set{Int64}()
+        disturbance_clusters::Set{Int64}=Set{Int64}(),
+        waypoint_optim_flag::Bool=true,
     )::MSTSolution
 
-Generate a solution to the problem for:
-- the mothership by using the nearest neighbour heuristic to generate an initial solution,
-    and then improving using the 2-opt heuristic, and
+Generate an initial solution to the problem for:
+- the mothership by using the nearest neighbour heuristic
+and then improving using the 2-opt heuristic, and
 -  for tenders using the sequential nearest neighbour heuristic.
+Optionally, optimize waypoints using gradient descent.
```

Comment:
Suggest adding a keyword argument that accepts an optimization method and then:

```suggestion
Optionally, optimize waypoints using a set or provided optimization method.
```